### PR TITLE
feat: check if build exists before deploying contract with pop up

### LINKED ISSUE #34: `pop up contract`: build contract if it has not been built
Based on this PR comment: https://github.com/r0gue-io/pop-cli/pull/27#discussion_r1515973155

For `pop up contract`, we should do a check, if the contract has not been built, then build it.

### PR DESCRIPTION
**Pull request by bolajahmad**
Closes #34 

`pop up contract` tries to deploy contract even if there is not build found and will eventually throw error because of that. It is necessary to add a check for existing build (and generate one), before continuing with the deployment.

### CODE REVIEW

File: crates/pop-cli/src/commands/up/contract.rs [hunk 1]
```diff
@@ -54,7 +54,23 @@ pub struct UpContractCommand {
...
 		clear_screen()?;
+
+		// Check if build exists in the specified "Contract build folder"
+		let build_path = PathBuf::from(
+			self.path.clone().unwrap_or("/.".into()).to_string_lossy().to_string()
+				+ "/target/release/ink",
+		);
+
+		if !build_path.exists() {
+			intro(format!("{}: Building a contract", style(" Pop CLI ").black().on_magenta()))?;
```
Comment (evilrobot-01) [inline crates/pop-cli/src/commands/up/contract.rs:65]: A `warning("NOTE: contract has not yet been built")` here before building would be nice. :)

Comment (bolajahmad) [reply crates/pop-cli/src/commands/up/contract.rs:65]: Oh, sounds good.

I don't understand the intent, should it be used alongside the current intro ("building a contract") or replace this altogether?

**Comment by AlexD10S** [comment]
If I don't have the contract build and run b`pop up contract` works perfectly.
However if I manually build the contract and then run `pop up contract` is still building it, for some reason it doesn't find the repo `!build_path.exists()`.

File: crates/pop-cli/src/commands/up/contract.rs (hunk 1 above)
Comment (bolajahmad) [reply crates/pop-cli/src/commands/up/contract.rs:68]: Alright, that makes sense!

**Comment by AlexD10S** [comment]
> If I don't have the contract build and run b`pop up contract` works perfectly. However if I manually build the contract and then run `pop up contract` is still building it, for some reason it doesn't find the repo `!build_path.exists()`.

The issue here is with this line:
```
let build_path = PathBuf::from(
			self.path.clone().unwrap_or("/.".into()).to_string_lossy().to_string()
				+ "/target/release/ink",
		);
```
`release` is not needed, check only for `target/ink`.
Also once this PR is implemented: https://github.com/r0gue-io/pop-cli/issues/102 and the user can specify if build in release mode or debug mode we will have to change it too.

**Comment by bolajahmad** [comment]
> > If I don't have the contract build and run b`pop up contract` works perfectly. However if I manually build the contract and then run `pop up contract` is still building it, for some reason it doesn't find the repo `!build_path.exists()`.
> 
> The issue here is with this line:
> 
> ```
> let build_path = PathBuf::from(
> 			self.path.clone().unwrap_or("/.".into()).to_string_lossy().to_string()
> 				+ "/target/release/ink",
> 		);
> ```
> 
> `release` is not needed, check only for `target/ink`. Also once this PR is implemented: #102 and the user can specify if build in release mode or debug mode we will have to change it too.

Alright, I'll try to remove the target/ink and see if that helps.

**Pull request merged**